Composer/packages/server/src/controllers/asset.ts

```diff
@@ -2,8 +2,11 @@
 // Licensed under the MIT License.
```

Comment:
A general ask for this file -- is all the data fetching in the controller layer the right thing?

Reply:
Okay I have exported most of these additions to AssetManager. I have left the readme function here since your other comment may lead to the removal of this technique.